src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Resources/Patch/FhirPatchInsertTests.cs

```diff
@@ -0,0 +1,155 @@
+                        new Identifier { System = "http://example.org", Value = "value 3" },
+                        new Identifier { System = "http://example.org", Value = "value 2" },
+                    },
+                }));
+        }
+
+        // Implements test case at:
+        // https://github.com/FHIR/fhir-test-cases/blob/752b01313ecbc1e13a942e1b3e25c96b3f7f3449/r5/patch/fhir-path-tests.xml#L733
+        [Fact]
+        public void GivenAFhirPatchInsertRequest_WhenInsertingStartOfList_ThenValueShouldExistAtStartOfList()
+        {
+            var patchParam = new Parameters().AddInsertPatchParameter("Patient.identifier", new Identifier { System = "http://example.org", Value = "value 3" }, 0);
+            var patientResource = new Patient
+            {
+                Identifier = new List<Identifier>
+                    {
+                        new Identifier { System = "http://example.org", Value = "value 1" },
+                        new Identifier { System = "http://example.org", Value = "value 2" },
+                    },
+            };
+
+            Patient patchedPatientResource = (Patient)new FhirPathPatchBuilder(patientResource, patchParam).Apply();
+
+            Assert.True(patchedPatientResource.Matches(
+                new Patient
+                {
+                    Identifier = new List<Identifier>
+                    {
+                        new Identifier { System = "http://example.org", Value = "value 3" },
+                        new Identifier { System = "http://example.org", Value = "value 1" },
+                        new Identifier { System = "http://example.org", Value = "value 2" },
+                    },
+                }));
+        }
+
+        [Fact]
+        public void GivenAFhirPatchInsertRequest_WhenInsertingInvalidPath_ThenInvalidOperationExceptionShouldBeThrown()
+        {
+            var patchParam = new Parameters().AddInsertPatchParameter("Patient.nothing", new FhirString("test"), 2);
+
+            var builder = new FhirPathPatchBuilder(new Patient(), patchParam);
+            Assert.Throws<InvalidOperationException>(builder.Apply);
+        }
+
+        // Not a defined test case, but the path will not resolve and thus a exception is expected.
+        // The FHIRPath must return a single element according to the spec and the path does not resolve.
+        [Fact]
+        public void GivenAFhirPatchInsertRequest_WhenInsertingToUninitializedList_ThenInvalidOperationExceptionShouldBeThrown()
+        {
+            var patchParam = new Parameters().AddInsertPatchParameter("Patient.identifier", new Identifier { System = "http://example.org", Value = "value 3" }, 2);
+
+            var builder = new FhirPathPatchBuilder(new Patient(), patchParam);
+            Assert.Throws<InvalidOperationException>(builder.Apply);
+        }
+
+        [Fact]
+        public void GivenAFhirPatchInsertRequest_WhenInsertingUninitializedList_ThenInvalidOperationExceptionShouldBeThrown()
```

Comment:
Nit: I think the test name is wrong. It looks like this test is for an index that is too high, not an uninitialized list.

Reply:
Test name was right, content misleading/wrong. Changed insert index to 0 to actually ensure the error we are getting matches the intended test case.